gen2-age-gender/main.py

```diff
@@ -74,36 +74,66 @@ def create_pipeline():
     # Script node will take the output from the face detection NN as an input and set ImageManipConfig
     # to the 'age_gender_manip' to crop the initial frame
     image_manip_script = pipeline.create(dai.node.Script)
-    image_manip_script.inputs['face_det_in'].setBlocking(False)
-    image_manip_script.inputs['face_det_in'].setQueueSize(4)
     face_det_nn.out.link(image_manip_script.inputs['face_det_in'])
+
+    # Only send metadata, we are only interested in timestamp, so we can sync
+    # depth frames with NN output
+    face_det_nn.passthrough.link(image_manip_script.inputs['passthrough'])
+
     image_manip_script.setScript("""
+l = [] # List of images
+# So the correct frame will be the first in the list
+# For this experiment this function is redundant, since everything
+# runs in blocking mode, so no frames will get lost
+def remove_prev_frame(seq):
+    for rm, frame in enumerate(l):
+        if frame.getSequenceNum() == seq:
+            # node.warn(f"List len {len(l)} Frame with same seq num: {rm},seq {seq}")
+            break
+    for i in range(rm):
```

Comment:
on the first iteration, if `l` is empty, the `rm` variable would not be initialized and throw error

Reply:
Thanks and thanks for noticing this! I'm not sure how inferencing could have arrived to the script node before the frame, but better to ba safe, especially when other users might edit this script.